packages/api/endpoints/collections.js

```diff
@@ -102,12 +110,14 @@ async function post(req, res) {
       throw e;
     }
   } catch (e) {
-    if (e.name === 'SchemaValidationError') {
-      return res.boom.badRequest(e.message);
-    }
-    if (e instanceof BadRequestError) {
+    if (
+      e.name === 'SchemaValidationError'
+      || e instanceof InvalidRegexError
+      || e instanceof UnmatchedRegexError
+    ) {
```

Comment:
I know this is not code that you've touched, but since you're in here, there are 2 other problems with the POST responses in this method:
1. line 93 should be changed from `notFound` to `badRequest` (400 is not appropriate because there was no attempt to find anything, rather the request is bad because it is missing information, and is thus badly formed)
2. line 99 should be changed from `badRequest` to `conflict` (409 [conflict] is standard practice for attempting to add something that already exists. 400 is not appropriate here because the request isn't "bad", meaning it is not an improperly formed request)